pkg/cmd/datastore/datastore.go

```diff
@@ -170,11 +178,16 @@ func RegisterDatastoreFlagsWithPrefix(flagSet *pflag.FlagSet, prefix string, opt
 	flagSet.StringVar(&opts.URI, flagName("datastore-conn-uri"), defaults.URI, `connection string used by remote datastores (e.g. "postgres://postgres:password@localhost:5432/spicedb")`)
 	flagSet.StringVar(&opts.CredentialsProviderName, flagName("datastore-credentials-provider-name"), defaults.CredentialsProviderName, fmt.Sprintf(`retrieve datastore credentials dynamically using (%s)`, datastore.CredentialsProviderOptions()))
 
+	flagSet.StringArrayVar(&opts.ReadReplicaURIs, flagName("datastore-read-replica-conn-uri"), []string{}, "connection string used by remote datastores for read replicas (e.g. \"postgres://postgres:password@localhost:5432/spicedb\")")
+	flagSet.StringVar(&opts.ReadReplicaCredentialsProviderName, flagName("datastore-read-replica-credentials-provider-name"), defaults.CredentialsProviderName, fmt.Sprintf(`retrieve datastore credentials dynamically using (%s)`, datastore.CredentialsProviderOptions()))
+	flagSet.BoolVar(&opts.ReadReplicaStrictMode, flagName("datastore-read-replica-strict-mode"), true, "enables strict mode for read replicas")
+
 	var legacyConnPool ConnPoolConfig
 	RegisterConnPoolFlagsWithPrefix(flagSet, "datastore-conn", DefaultReadConnPool(), &legacyConnPool)
 	deprecateUnifiedConnFlags(flagSet)
 	RegisterConnPoolFlagsWithPrefix(flagSet, "datastore-conn-pool-read", &legacyConnPool, &opts.ReadConnPool)
 	RegisterConnPoolFlagsWithPrefix(flagSet, "datastore-conn-pool-write", DefaultWriteConnPool(), &opts.WriteConnPool)
+	RegisterConnPoolFlagsWithPrefix(flagSet, "datastore-read-replica-conn-pool", DefaultReadConnPool(), &opts.ReadReplicaConnPool)
```

Comment:
I wanted all the read replica flags to start with `datastore-read-replica` to be explicit

Reply:
I can tell you that as someone usually tinkering with these flags, having to remember 2 different naming patterns is not going to make my life easier

Reply:
Its a tradeoff, but I still think having all the read replica flags sharing a common prefix makes more sense. Perhaps we should ask?

Reply:
what is the tradeoff?

Reply:
Its harder discover using autocomplete: right now I can do `--datastore-read-replica-` and tab to see *all* the read replica flags

Reply:
@jzelinskie can you share your thoughts on the flag UX here?

Reply:
A few questions/thoughts that could help us understand the usage here:
- When would you ever want to configure read replica pool to be not the same as the read pool?
- Do we think we could use a heuristic inside SpiceDB to come up with these values rather than even exposing them at all?

Reply:
> When would you ever want to configure read replica pool to be not the same as the read pool?

Always: you can't use the read pool with replicas

> Do we think we could use a heuristic inside SpiceDB to come up with these values rather than even exposing them at all?

Incredibly unlikely... We need to know that a replica is a replica

Reply:
>Always: you can't use the read pool with replicas

Configure them differently, not use the same pool.
For example, if I'm using a RDS with a read replica, the read replica is likely the same size as my primary likely with the same max connections.

>Incredibly unlikely... We need to know that a replica is a replica

I'm not advocating altering the system to be unaware of read replicas. I'm advocating for the system being smart enough to discover the right size of connection pools because users are very likely to pick non-ideal values themselves

Reply:
> For example, if I'm using a RDS with a read replica, the read replica is likely the same size as my primary likely with the same max connections.

Which still means the read pool for non-replicas will want to be configured differently, as we also have a write pool for the primary.

>  I'm advocating for the system being smart enough to discover the right size of connection pools because users are very likely to pick non-ideal values themselves

And... how would it do that?

Reply:
Do you think a simplified version of [adaptive concurrency limits](https://netflixtechblog.medium.com/performance-under-load-3e6fa9a60581) (e.g. similar to tcp congestion control) would work? Basically use connection errors to adaptively discover the right number of connections for you pool -- something like this could even make the system robust to changes in the replicas or the size of the database.

Reply:
Perhaps, but it is IMO outside of the scope of this change (as it would apply to all pools, not just replicas), and would require a significant amount of testing.